metrics/cli/docs/02_metrics_cli_firebase_analytics_investigation.md

```diff
@@ -0,0 +1,80 @@
+
+Cons:
+
+- not an automatic solution.
+
+### Using an API
+
+The second solution is to use a [Firebase Management API](https://firebase.google.com/docs/projects/api/reference/rest) and a [Google Analytics Management API](https://developers.google.com/analytics/devguides/config/mgmt/v3/mgmtReference).
+
+The APIs contain the following required endpoints:
+- [`Google analytics summaries`](https://developers.google.com/analytics/devguides/config/mgmt/v3/mgmtReference/management/accountSummaries/list) to retrieve a Google analytics' account id;
+- [`Add google analytics`](https://firebase.google.com/docs/projects/api/reference/rest/v1beta1/projects/addGoogleAnalytics) to link the Google analytics with Firebase Analytics.
+    
+To enable Firebase analytics using the API described above, consider the following steps:
+
+1. Enable Google analytics API in GCloud, which create Google analytics account for the newly created GCloud project.
+2. Call the `Google analytics summaries` endpoint.
+3. Call the `Add google analytics` endpoint with the Google analytics' account id retrieved from the previous step.
+
+Unfortunately, the `Google analytics summaries` endpoint require a token with an `analytics.edit` scope. 
+
+Here is a table, which describes how to get the token with a specific scope:
+
+| Solution | **Description** | **Problems** |
+| --- | --- | --- |
+| **[Configure OAuth 2.0](https://support.google.com/cloud/answer/6158849) for the current GCloud project** | Allows us to get the token with required scope using OAuth 2.0 | The user should manually configure an [OAuth consent screen](https://support.google.com/cloud/answer/10311615) |
+| **Enable [compute engine API](https://cloud.google.com/compute/docs/reference/rest/v1) in GCloud account** | Allow us to give a required scope for Gcloud project service accounts using the command described [here](https://cloud.google.com/compute/docs/access/create-enable-service-accounts-for-instances). Then we can [authenticate](https://cloud.google.com/sdk/gcloud/reference/auth/activate-service-account) using the service account with the required scope and [print the service account token](https://cloud.google.com/sdk/gcloud/reference/auth/print-access-token). | The compute engine API requires enabling the Gcloud billing account, which a manually step only |
+
+Let's describe the pros and cons of the API solution.
+
+Pros:
+
+- perhaps in the future, it will be possible to enable the OAuth consent screen or billing account programmatically.
+
+Cons: 
+
+- complicated implementation;
+- requires complicated steps from the user side (configure OAuth consent screen, configure billing account);
+- for now, it is not an automatic solution.
+
+## Decision
+
+As we've analyzed above, the [API solution](#using-an-api) is too complicated for now. We should choose a described [manual solution](#manual) to enable Firebase Analytics.
```

Comment:
I think we've chosen the manual option because the API does not allow us to automate the Firebase Analytics enabling process, and not because this solution is too complex.